Firestore/core/src/firebase/firestore/core/firestore_client.mm

```diff
@@ -159,39 +153,39 @@ new FirestoreClient(database_info, std::move(credentials_provider),
   // more work) since external write/listen operations could get queued to run
   // before that subsequent work completes.
   if (settings.persistence_enabled()) {
-    Path dir = [FSTLevelDB
-        storageDirectoryForDatabaseInfo:database_info_
-                     documentsDirectory:[FSTLevelDB documentsDirectory]];
+    Path dir = LevelDbPersistence::StorageDirectory(
+        database_info_, LevelDbPersistence::DocumentsDirectory());
 
     FSTSerializerBeta* remote_serializer = [[FSTSerializerBeta alloc]
         initWithDatabaseID:database_info_.database_id()];
     FSTLocalSerializer* serializer =
         [[FSTLocalSerializer alloc] initWithRemoteSerializer:remote_serializer];
-    FSTLevelDB* ldb;
-    Status level_db_status = [FSTLevelDB
-        dbWithDirectory:std::move(dir)
-             serializer:serializer
-              lruParams:LruParams::WithCacheSize(settings.cache_size_bytes())
-                    ptr:&ldb];
-    if (!level_db_status.ok()) {
+
+    auto created = LevelDbPersistence::Create(
+        std::move(dir), serializer,
+        LruParams::WithCacheSize(settings.cache_size_bytes()));
+    if (!created.ok()) {
       // If leveldb fails to start then just throw up our hands: the error is
       // unrecoverable. There's nothing an end-user can do and nearly all
       // failures indicate the developer is doing something grossly wrong so we
       // should stop them cold in their tracks with a failure they can't ignore.
       [NSException
```

Comment:
Seems like something we'll have to hide behind an abstraction layer (not in this PR)?

Reply:
Absolutely. No todo required because of the overarching work of getting this file to end in .cc.